dotnet/maui
# Added pending changes to Essentials Permissions from Xamarin.Forms

### Description of Change

Added pending changes to Essentials Permissions from Xamarin.Forms. Doing triage we noticed that this changes https://github.com/xamarin/Essentials/pull/1937 were pending to include in .NET MAUI.

To test the changes:
1. Call ` await Permissions.RequestAsync<Permissions.LocationWhenInUse>();`, which shows a popup asking whether you allow approximate location or precise location or not allow location at all.
2. Select approximate location-
3. See the return value is `PermissionStatus.Allowed`.

### Issues Fixed

Fixes #10053
